### PR TITLE
Test for a set operation on an unreachable device

### PR DESCRIPTION
**Pull request by ray-milkey**
This PR adds an integration test to check that a write to a device that cannot be reached completes properly and sets the cache properly.

### CODE REVIEW

File: test/gnmi/unreachabledevicetest.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	// Make a GNMI client to use for requests
+	c, err := env.Config().NewGNMIClient()
+	assert.NoError(t, err)
+	assert.True(t, c != nil, "Fetching client returned nil")
+
+	// Set a value using gNMI client to the device
+	extNameDeviceType := gnmi_ext.Extension_RegisteredExt{
+		RegisteredExt: &gnmi_ext.RegisteredExtension{
+			Id:  gnmi.GnmiExtensionDeviceType,
+			Msg: []byte("Devicesim"),
```
Comment (SeanCondon) [inline test/gnmi/unreachabledevicetest.go]: you could reuse `unreachableDeviceModDeviceType` defined at the top

File: test/gnmi/unreachabledevicetest.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			Msg: []byte(unreachableDeviceModDeviceVersion),
+		},
+	}
+	extensions := []*gnmi_ext.Extension{{Ext: &extNameDeviceType}, {Ext: &extNameDeviceVersion}}
+
+	setPath := makeDevicePath(unreachableDeviceModDeviceName, unreachableDeviceModPath)
+	setPath[0].pathDataValue = unreachableDeviceModValue
+	setPath[0].pathDataType = StringVal
+
+	_, extensionsSet, errorSet := gNMISet(MakeContext(), c, setPath, noPaths, extensions)
```
Comment (SeanCondon) [inline test/gnmi/unreachabledevicetest.go]: As i understand it - this will not return - it should do, but we're waiting on Jordan's patch

Comment (ray-milkey) [reply test/gnmi/unreachabledevicetest.go]: Correct. This cannot go in until the other changes are merged.

**Comment by ray-milkey** [comment]
This one works with @kuujo change to decouple NB and SB

**Pull request merged**